File: src/main/java/com/ddd/trainings/infra/TrainingIdeaRepo.java
package com.ddd.trainings.infra;

import com.ddd.trainings.domain.IdeaId;
import com.ddd.trainings.domain.TrainingIdea;
import java.util.List;
import java.util.Optional;

public interface TrainingIdeaRepo {

  TrainingIdea save(TrainingIdea idea);

  List<TrainingIdea> findAll();

  Optional<TrainingIdea> findById(IdeaId ideaId);
}

File: src/Ddd/Trainings/Idea/Infra/TrainingIdeaRepo.cs
using System.Collections.Generic;
using Ddd.Trainings.Idea.Domain;

namespace Ddd.Trainings.Idea.Infra
{
    public interface ITrainingIdeaRepo
    {
        TrainingIdea Save(TrainingIdea idea);
        List<TrainingIdea> FindAll();
        TrainingIdea FindById(IdeaId ideaId);
    }
} 